pkg/apis/messaging/v1alpha1/pipeline_lifecycle.go

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2019 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+import duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+
+const (
+	// PipelineConditionReady has status True when all subconditions below have been set to True.
+	PipelineConditionReady = duckv1alpha1.ConditionReady
+
+	// PipelineConditionReferencesResolved has status True when all the specified references have been successfully
+	// resolved.
+	PipelineConditionReferencesResolved duckv1alpha1.ConditionType = "Resolved"
+
+	// PipelineConditionChannelReady has status True when controller has successfully added a
+	// pipeline to the spec.channel resource.
+	PipelineConditionChannelReady duckv1alpha1.ConditionType = "ChannelReady"
```

Comment:
Does this represent all Channels used in the pipeline? Also, the comment about `spec.channel` doesn't make sense to me.

Reply:
Sorry, there's left over cruft, please just look at the pipeline_types.go file. Sorry, tried to direct attention to ONLY that file :)